### ISSUE TITLE
Attempted to load class "Validator" from namespace "JsonSchema"

### ISSUE THREAD

**Issue by cacahouete78** [phiamo/MopaBootstrapBundle#1112]
Since a week, there are the following error when I try to use a command like mopa:bootstrap:install:font, mopa:bootstrap:symlink:less, mopa:bootstrap:symlink:sass.

```
Fatal error: Class 'JsonSchema\Validator' not found in phar://D:/Projects/test/composer.phar/src/Composer/Json/JsonFile.php on line 160

Call Stack:
    0.0003     239176   1. {main}() D:\Projects\test\app\console:0
    0.4295    3869808   2. Symfony\Component\Console\Application->run() D:\Projects\test\app\console:27
    0.4979    4110552   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() D:\Projects\test\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:126
    3.8588   18913360   4. Symfony\Component\Console\Application->doRun() D:\Projects\test\vendor\symfony\symfony\src\Symfony\Bundle\FrameworkBundle\Console\Application.php:96
    3.8596   18914232   5. Symfony\Component\Console\Application->doRunCommand() D:\Projects\test\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:195
    3.9412   19148272   6. Symfony\Component\Console\Command\Command->run() D:\Projects\test\vendor\symfony\symfony\src\Symfony\Component\Console\Application.php:886
    3.9422   19152792   7. Mopa\Bundle\BootstrapBundle\Command\InstallFontCommand->execute() D:\Projects\test\vendor\symfony\symfony\src\Symfony\Component\Console\Command\Command.php:259
    3.9508   19193224   8. Mopa\Bridge\Composer\Adapter\ComposerAdapter::getComposer() D:\Projects\test\vendor\mopa\bootstrap-bundle\Mopa\Bundle\BootstrapBundle\Command\InstallFontCommand.php:67
    4.1001   19399672   9. Mopa\Bridge\Composer\Adapter\ComposerAdapter::createComposer() D:\Projects\test\vendor\mopa\composer-bridge\src\Mopa\Bridge\Composer\Adapter\ComposerAdapter.php:114
    4.1025   19655648  10. Composer\Factory::create() D:\Projects\test\vendor\mopa\composer-bridge\src\Mopa\Bridge\Composer\Adapter\ComposerAdapter.php:69
    4.1025   19656656  11. Composer\Factory->createComposer() phar://D:/Projects/test/composer.phar/src/Composer/Factory.php:490
    4.1046   19862832  12. Composer\Json\JsonFile->validateSchema() phar://D:/Projects/test/composer.phar/src/Composer/Factory.php:222

[2015-09-21 15:06:33] php.CRITICAL: Fatal Error: Class 'JsonSchema\Validator' not found {"type":1,"file":"phar://D:/Projects/test/composer.phar/src/Composer/Json/JsonFile.php","line":160,"level":-1,"stack":[{"function":"validateSchema","type":"->","class":"Composer\\Json\\JsonFile","file":"phar://D:/Projects/test/composer.phar/src/Composer/Factory.php","line":222,"args":[]},{"function":"createComposer","type":"->","class":"Composer\\Factory","file":"phar://D:/Projects/test/composer.phar/src/Composer/Factory.php","line":490,"args":[]},{"function":"create","type":"::","class":"Composer\\Factory","file":"D:\\Projects\\test\\vendor\\mopa\\composer-bridge\\src\\Mopa\\Bridge\\Composer\\Adapter\\ComposerAdapter.php","line":69,"args":[]},{"function":"createComposer","type":"::","class":"Mopa\\Bridge\\Composer\\Adapter\\ComposerAdapter","file":"D:\\Projects\\test\\vendor\\mopa\\composer-bridge\\src\\Mopa\\Bridge\\Composer\\Adapter\\ComposerAdapter.php","line":114,"args":[]},{"function":"getComposer","type":"::","class":"Mopa\\Bridge\\Composer\\Adapter\\ComposerAdapter","file":"D:\\Projects\\test\\vendor\\mopa\\bootstrap-bundle\\Mopa\\Bundle\\BootstrapBundle\\Command\\InstallFontCommand.php","line":67,"args":[]},{"function":"execute","type":"->","class":"Mopa\\Bundle\\BootstrapBundle\\Command\\InstallFontCommand","file":"D:\\Projects\\test\\vendor\\symfony\\symfony\\src\\Symfony\\Component\\Console\\Command\\Command.php","line":259,"args":[]},{"function":"run","type":"->","class":"Symfony\\Component\\Console\\Command\\Command","file":"D:\\Projects\\test\\vendor\\symfony\\symfony\\src\\Symfony\\Component\\Console\\Application.php","line":886,"args":[]},{"function":"doRunCommand","type":"->","class":"Symfony\\Component\\Console\\Application","file":"D:\\Projects\\test\\vendor\\symfony\\symfony\\src\\Symfony\\Component\\Console\\Application.php","line":195,"args":[]},{"function":"doRun","type":"->","class":"Symfony\\Component\\Console\\Application","file":"D:\\Projects\\test\\vendor\\symfony\\symfony\\src\\Symfony\\Bundle\\FrameworkBundle\\Console\\Application.php","line":96,"args":[]},{"function":"doRun","type":"->","class":"Symfony\\Bundle\\FrameworkBundle\\Console\\Application","file":"D:\\Projects\\test\\vendor\\symfony\\symfony\\src\\Symfony\\Component\\Console\\Application.php","line":126,"args":[]},{"function":"run","type":"->","class":"Symfony\\Component\\Console\\Application","file":"D:\\Projects\\test\\app\\console","line":27,"args":[]},{"function":"{main}","file":"D:\\Projects\\test\\app\\console","line":0,"args":[]}]} 


                                                                               
  [Symfony\Component\Debug\Exception\ClassNotFoundException]                   
  Attempted to load class "Validator" from namespace "JsonSchema".             
  Did you forget a "use" statement for e.g. "Gedmo\ReferenceIntegrity\Mapping  
  \Validator", "Gedmo\SoftDeleteable\Mapping\Validator", "Gedmo\Tree\Mapping\  
  Validator", "Gedmo\Uploadable\Mapping\Validator" or "Symfony\Component\Vali  
  dator\Validator"?                                                            
                                                                               


mopa:bootstrap:install:font
```

Somebody have a solution or can you fix this problem ?

**Comment by isometriks** [comment]
Hmm, seems to be a composer issue or something. https://github.com/composer/composer/blob/master/src/Composer/Json/JsonFile.php#L160 might want to try a composer self-update?

**Comment by cacahouete78** [comment]
Hi, I "composer self-update" each morning before "composer update". MOPA's commands are the only ones i can't run.

My php client and Bundles :
```
PHP 5.5.29 (cli) (built: Sep  2 2015 16:48:08)
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
    with Xdebug v2.3.3, Copyright (c) 2002-2015, by Derick Rethans

behat/transliterator                     v1.0.1            
cocur/slugify                            v1.3              
craue/formflow-bundle                    2.1.8             
doctrine/annotations                     v1.2.7            
doctrine/cache                           v1.4.2            
doctrine/collections                     v1.3.0            
doctrine/common                          v2.5.1            
doctrine/dbal                            v2.5.2            
doctrine/doctrine-bundle                 v1.5.1            
doctrine/doctrine-cache-bundle           v1.0.1            
doctrine/inflector                       v1.0.1            
doctrine/instantiator                    1.0.5             
doctrine/lexer                           v1.0.1            
doctrine/orm                             v2.5.1            
egeloen/ckeditor-bundle                  2.5.2             
egeloen/json-builder                     1.0.3             
friendsofsymfony/rest-bundle             1.7.1             
friendsofsymfony/user-bundle             v1.3.6            
gedmo/doctrine-extensions                v2.4.6            
genemu/form-bundle                       v2.2.2            
guzzle/guzzle                            v3.9.3            
imagine/imagine                          0.6.2             
incenteev/composer-parameter-handler     v2.1.1            
jdorn/sql-formatter                      v1.2.17           
jms/metadata                             1.5.1             
jms/parser-lib                           1.0.0             
jms/serializer                           1.0.0             
jms/serializer-bundle                    1.0.0             
knplabs/gaufrette                        v0.1.9            
knplabs/knp-components                   1.3.2             
knplabs/knp-markdown-bundle              1.3.2             
knplabs/knp-menu                         v2.1.0            
knplabs/knp-menu-bundle                  v2.0.0            
knplabs/knp-paginator-bundle             2.5.0             
knplabs/knp-time-bundle                  1.4.0             
kriswallsmith/assetic                    v1.3.0            
kriswallsmith/buzz                       v0.15             
michelf/php-markdown                     1.4.0             
monolog/monolog                          1.17.1            
mopa/bootstrap-bundle                    dev-master cc093b4
mopa/composer-bridge                     v1.4.0            
nelmio/api-doc-bundle                    2.9.0             
phpcollection/phpcollection              0.4.0             
phpoption/phpoption                      1.5.0             
presta/sitemap-bundle                    v1.4.0            
psr/log                                  1.0.0             
sensio/distribution-bundle               v3.0.31           
sensio/framework-extra-bundle            v3.0.10           
sensio/generator-bundle                  v2.5.3            
sensiolabs/security-checker              v3.0.1
sonata-project/admin-bundle              dev-master b60b81b
sonata-project/block-bundle              2.3.8             
sonata-project/cache                     1.0.6             
sonata-project/classification-bundle     dev-master ddf354c
sonata-project/core-bundle               2.3.9             
sonata-project/datagrid-bundle           dev-master cc42f28
sonata-project/doctrine-extensions       1.0.2             
sonata-project/doctrine-orm-admin-bundle dev-master dbcd559
sonata-project/easy-extends-bundle       2.1.9             
sonata-project/exporter                  1.4.0             
sonata-project/formatter-bundle          dev-master 84ae2db
sonata-project/google-authenticator      1.0.2             
sonata-project/intl-bundle               dev-master c50bda8
sonata-project/media-bundle              dev-master 7456289
sonata-project/notification-bundle       2.3.2             
sonata-project/seo-bundle                2.0.0             
sonata-project/user-bundle               dev-master ec09378
swiftmailer/swiftmailer                  v5.4.1            
symfony/assetic-bundle                   v2.7.0            
symfony/monolog-bundle                   v2.7.1            
symfony/swiftmailer-bundle               v2.3.8            
symfony/symfony                          v2.7.4            
twbs/bootstrap                           dev-master 911312d
twbs/bootstrap-sass                      dev-master f1d9a24
twig/extensions                          v1.3.0            
twig/twig                                v1.22.1           
willdurand/jsonp-callback-validator      v1.1.0            
willdurand/negotiation                   1.4.0 
```

**Comment by aashley** [comment]
mopa composer adapptor is building its own autoloader and not supporting PSR-4 properly see https://github.com/phiamo/MopaComposerBridge/issues/15

**Comment by ttiu** [comment]
@cacahouete78 The last tagged version of composer could be a temporary fix.
````
composer self-update  1.0.0-alpha10
````

**Comment by phiamo** [comment]
Can you try a composer update, i released a new version of composer bridge containing a fix regarding issue #1112
